packages/mobile/src/components/user/FollowButton.tsx

```diff
@@ -28,11 +28,13 @@ export const FollowButton = (props: FollowButtonsProps) => {
   const handlePress = useCallback(
     (event: GestureResponderEvent) => {
       onPress?.(event)
-      if (isFollowing) {
-        dispatch(unfollowUser(userId, followSource))
-      } else {
-        dispatch(followUser(userId, followSource))
-      }
+      setTimeout(() => {
```

Comment:
i assume we should be using the requestanimationframe equivalent for rn-native right @amendelsohn ?

Reply:
oh yeah that works too

Reply:
I don't think so. RAF will get us time on the thread before the next paint, but the whole problem here is that we're doing too much computation between renders and so the animation isn't playing out fast enough. RAF helps block the main thread for quick important tasks like updating the progress bar offset, but putting an expensive callback there will just lengthen the time before the next render.

If our callback was lighter weight or had less downstream impact, we might be able to do that, but I think it's likely not helpful here.

There is the opposite in `requestIdleCallback` which specifically looks for idle time to execute the callback. However, this is web only and we'd have to use an alternative like `@shopify/react-idle`

Reply:
It looks like RAF is working fine in this case, but I'm still not convinced it's conceptually correct. Possible that this callback is lightweight enough that it's equivalent to setTimeout.

Down to ship as is and circle back though since perf looks good

Reply:
Oh yeah I see what you mean, sg

Reply:
Yeah I also had a backwards understanding of RAF before